tests/test-sources/plugins/git/octo.nix

```diff
@@ -0,0 +1,23 @@
+{
+  empty = {
+    plugins.octo.enable = true;
+  };
+
+  example = {
+    plugins.octo = {
+      enable = true;
+
+      settings = {
+        ssh_aliases = {
+          "github.com-work" = "github.com";
+        };
+        mappings_disable_default = true;
+        mappings = {
+          issue.react_heart = "<leader>rh";
+          file_panel.select_prev_entry = "[q";
+        };
+        picker = "fzf-lua";
```

Comment:
This line seems to be causing the [CI failure](https://buildbot.nix-community.org/#/builders/165/builds/346) (Error loading picker provider fzf-lua):

```
plugins-git-octo-example> Running phase: patchPhase
plugins-git-octo-example> Running phase: updateAutotoolsGnuConfigScriptsPhase
plugins-git-octo-example> Running phase: configurePhase
plugins-git-octo-example> no configure script, doing nothing
plugins-git-octo-example> Running phase: buildPhase
plugins-git-octo-example> ERROR: Error detected while processing /nix/store/6rj1b1syw12nkgh15g13qqdkivqs1mii-init.lua:
plugins-git-octo-example> Error loading picker provider fzf-lua
plugins-git-octo-example> Cannot request projects v2, missing scope 'read:project'
error: builder for '/nix/store/d83f3vcjkinm2acriiqlkfp1yaizzgak-plugins-git-octo-example.drv' failed with exit code 1
```

Reply:
yeah. saw that. since I don't provide it as an option, I have changed it to telescope. should I add it as an option and install it if it's selected (with an option to disable install)?

Reply:
I think the error was less to do with a nix option not existing and more that the selected picker isn't available at runtime.

I'd suggest moving the `picker` test into a separate `testWithPicker` test and also add the dependency via `extraPackages`.

---

We _could_ consider having a "smart" way of configuring a picker, but IDK if there's a deterministic way to map a string accepted by the `picker` option to an actual dependency package?

It'd also have to be an `extraOptions` rather than a `settingsOptions`, because the latter is reserved for stuff options that we essentially blindly pass to `toLuaObject` without attempting to be too clever.

Reply:
> I think the error was less to do with a nix option not existing and more that the selected picker isn't available at runtime.

yeah. i meant that as I didn't provide an option for the picker that would install fzf

> but IDK if there's a deterministic way to map a string accepted by the picker option to an actual dependency package?

picker would be an enum of telescope and fzf-lua, so it wouldn't be that hard [src](https://github.com/pwntester/octo.nvim?tab=readme-ov-file#-configuration).

still don't think we need to add it though.

Reply:
> picker would be an enum of telescope and fzf-lua

In that case I think we should probably add the option, even if it's just for documentation purposes.

We could either note in the description that "it is the user's responsibility to ensure the picker is installed, e.g. via `extraPackages` or enabling the relevant plugin" or we could handle it automatically in `extraConfig`.

While as a rule we try and be fairly hands off with settings-options, we do still set sane defaults and/or upstream-recommended defaults when it makes sense.

Thoughts @GaetanLepage?

Reply:
If the mapping to package is "obvious" I think we should handle it